sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -604,6 +604,18 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
     }
   }
 
+  test("sort should not be introduced when aliases are used") {
+    withSQLConf(SQLConf.AUTO_BROADCASTJOIN_THRESHOLD.key -> "0") {
+      withTable("t") {
+        df1.repartition(1).write.format("parquet").bucketBy(8, "i").sortBy("i").saveAsTable("t")
```

Comment:
Could you add more tests, e.g., orderBy cases?

Reply:
Do you mean something like the following?
```
val t1 = spark.range(10).selectExpr("id as k1").orderBy("k1")
  .selectExpr("k1 as k11").orderBy("k11")
val t2 = spark.range(10).selectExpr("id as k2").orderBy("k2")
t1.join(t2, t1("k11") === t2("k2")).explain(true)
```
Extra `Sort` is optimized away, so it doesn't affect the physical plan (no extra sort) without this change.

Reply:
Ah, I see. How about the aggregate case?

Reply:
I added a test for `SortAggregateExec`. `HashAggregateExec` and `ObjectAggregateExec` are not affected since ordering is not involved with them.